File: app/src/main/java/fr/piotr/dismoitoutsms/contacts/ContactsAdapter.java
package fr.piotr.dismoitoutsms.contacts;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import fr.piotr.dismoitoutsms.R;
import fr.piotr.dismoitoutsms.util.ContactHelper;

public class ContactsAdapter extends ArrayAdapter<Contact> {

	private Context context;

	public ContactsAdapter(Context context, Contacts contacts) {
		super(context, R.layout.contact, contacts.asList());
		this.context = context;
	}

	public static class ContactViewHolder {
		private ImageView avatar;
		private ImageView	photo;
		private TextView	nom;

		public ContactViewHolder() {
			//
		}

		public ContactViewHolder(ImageView avatar, ImageView	photo, TextView nom) {
			this.avatar = avatar;
			this.photo = photo;
			this.nom = nom;
		}

		public ImageView getAvatar() {
			return avatar;
		}

		public ImageView getPhoto() {
			return photo;
		}

		/**
		 * @return the nom
		 */
		public TextView getNom() {
			return nom;
		}

	}

	@NonNull
	@Override
	public View getView(final int position, View convertView, @NonNull ViewGroup parent) {
		final Contact contact = getItem(position);

		ImageView avatar;
		ImageView photo;
		final TextView textViewNom;
		if (convertView == null) {

			convertView = LayoutInflater.from(context).inflate(R.layout.contact, parent, false);
			avatar = convertView.findViewById(R.id.avatar);
			photo = convertView.findViewById(R.id.photoContact);
			textViewNom = convertView.findViewById(R.id.nom);

			convertView.setTag(new ContactViewHolder(avatar, photo, textViewNom));

		} else {
			ContactViewHolder holder = (ContactViewHolder) convertView.getTag();
			avatar = holder.getAvatar();
			photo = holder.getPhoto();
			textViewNom = holder.getNom();
		}

		if (contact.hasAPhoto()) {
			photo.setVisibility(View.VISIBLE);
			photo.setImageBitmap(ContactHelper.getPhotoContact(context, contact.getPhotoId()));
			avatar.setVisibility(View.GONE);
		} else {
			photo.setVisibility(View.GONE);
			avatar.setVisibility(View.VISIBLE);
		}
		textViewNom.setText(contact.getName());

		return convertView;
	}

}

File: app/src/main/java/fr/piotr/dismoitoutsms/contacts/ContactsAdapter.kt
package fr.piotr.dismoitoutsms.contacts

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

import fr.piotr.dismoitoutsms.R
import fr.piotr.dismoitoutsms.util.ContactHelper

class ContactsAdapter(mContext: Context, contacts: Contacts) : ArrayAdapter<Contact>(mContext, R.layout.contact, contacts.asList()) {

    data class ContactViewHolder(var avatar: ImageView, var photo: ImageView, var nom: TextView)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var mConvertView = convertView
        val contact = getItem(position)

        val avatar: ImageView
        val photo: ImageView
        val textViewNom: TextView
        if (mConvertView == null) {

            mConvertView = LayoutInflater.from(context).inflate(R.layout.contact, parent, false)
            avatar = mConvertView.findViewById(R.id.avatar)
            photo = mConvertView.findViewById(R.id.photoContact)
            textViewNom = mConvertView.findViewById(R.id.nom)

            mConvertView.tag = ContactViewHolder(avatar, photo, textViewNom)

        } else {
            val holder = mConvertView.tag as ContactViewHolder
            avatar = holder.avatar
            photo = holder.photo
            textViewNom = holder.nom
        }

        if (contact!!.hasAPhoto()) {
            photo.visibility = View.VISIBLE
            photo.setImageBitmap(ContactHelper.getPhotoContact(context, contact.photoId))
            avatar.visibility = View.GONE
        } else {
            photo.visibility = View.GONE
            avatar.visibility = View.VISIBLE
        }
        textViewNom.text = contact.name

        return mConvertView
    }

}
